CONTRIBUTING.md

```diff
@@ -1,71 +1,73 @@
 
-1. Check first if your issue has already been [reported](https://github.com/ocaml-ppx/ocamlformat/issues/).
-2. Include the version of OCamlFormat you are using (`ocamlformat --version`).
-3. Include a (preferably short) source file which demonstrates the issue.
-4. Describe the expected and actual behavior.
-5. Do not unsubscribe from the issue until it is closed, the maintainers may ask for your feedback.
+1. First, verify if your issue hasn't already been [reported](https://github.com/ocaml-ppx/ocamlformat/issues/).
+2. Specify the OCamlFormat version you're using (`ocamlformat --version`).
+3. Include a (preferably short) source file that replicates the issue.
+4. Clearly define the expected and actual behavior.
+5. Remain engaged with the issue until closure as your feedback might be needed.
 
-When acknowledged, the project maintainers will add [labels](#ocamlformat-labels) to your issue throughout its lifespan.
+Once your issue is acknowledged, our maintainers will apply relevant [labels](#ocamlformat-labels) to track its status.
 
 ## Pull Requests
 
-We actively welcome pull requests.
+We heartily welcome pull requests. To ensure an effective contribution, please adhere to the steps below:
 
-1. Prior to investing a large amount of time into significant or invasive changes, or those that affect the output, it is likely more efficient to first open an issue for discussion and planning.
-2. If you are not familiar with the project, focus first on the [Good first issues](https://github.com/ocaml-ppx/ocamlformat/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22Good-first-issue+%3A%2B1%3A%22).
+1. For significant, invasive, or output-affecting changes, consider opening an issue for discussion before committing substantial time and effort.
+2. If you're new to the project, starting with the [Good first issues](https://github.com/ocaml-ppx/ocamlformat/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22Good-first-issue+%3A%2B1%3A%22) can be beneficial.
 3. Fork the repository and create your branch from `main`.
-4. If you have added code that should be tested, add tests (they should be located in the `tests/passing` directory).
-'.
-5. Ensure the test suite passes (see [Running the tests](#running-the-tests)).
-
-When acknowledged, the project maintainers will add [labels](#ocamlformat-labels) to your pull request throughout its lifespan.
+4. If you've added code that should be tested, supplement it with tests located in the `tests/passing` directory.
+5. Ensure that the test suite passes (see [Running the tests](#running-the-tests) for instructions).
 
+After acknowledgment, maintainers will tag your pull request with appropriate [labels](#ocamlformat-labels).
 
-### Running the tests
+### Running the Tests
 
-Once OCamlFormat has been built, run `make test` to check for regressions.
+After building OCamlFormat, execute `make test` to check for any regressions. Here's what happens during `make test`:
```

Comment:
The Makefile is useful for documentation but it doesn't have to be used. This is `dune runtest`.

I suggest always running `dune runtest --auto-promote` for the following advantages:
- Use of the better diff-viewer that people might have in their editor to view the formatting changes
- The state of the tests always reflect the code

Almost all tests will promote back to their original state if the code is reverted and run again.